Kitodo-Docket/src/main/java/org/kitodo/docket/ExportXmlLog.java

```diff
@@ -282,7 +305,119 @@ private List<Element> prepareProperties(List<Property> properties, Namespace xml
+                config.setReloadingStrategy(new FileChangedReloadingStrategy());
+
+                int count = config.getMaxIndex("namespace");
+                for (int i = 0; i <= count; i++) {
+                    String name = config.getString("namespace(" + i + ")[@name]");
+                    String value = config.getString("namespace(" + i + ")[@value]");
+                    nss.put(name, value);
+                }
+            }
+        } catch (ConfigurationException | RuntimeException e) {
+            logger.debug(e.getMessage(), e);
+            nss = new HashMap<>();
+        }
+        return nss;
+    }
+
+    private static void prepareMetadataElements(List<Element> metadataElements, boolean useAnchor, DocketData docketData,
+            Namespace[] namespaces, Namespace xmlns)
+            throws IOException, JDOMException {
+        HashMap<String, String> fields = getMetsFieldsFromConfig(useAnchor);
+        Document metsDoc = new SAXBuilder().build(docketData.metadataFile());
+        prepareMetadataElements(metadataElements, fields, metsDoc, namespaces, xmlns);
+    }
+
+    private static void prepareMetadataElements(List<Element> metadataElements, Map<String, String> fields, Document document,
+            Namespace[] namespaces, Namespace xmlns) {
+        for (Map.Entry<String, String> entry : fields.entrySet()) {
+            String key = entry.getKey();
+            List<Element> metsValues = getMetsValues(entry.getValue(), document, namespaces);
+            for (Element element : metsValues) {
+                Element ele = new Element(PROPERTY, xmlns);
+                ele.setAttribute("name", key);
+                ele.addContent(element.getTextTrim());
+                metadataElements.add(ele);
+            }
+        }
+    }
+
+    private static HashMap<String, String> getMetsFieldsFromConfig(boolean useAnchor) {
+        String xmlpath = "mets." + PROPERTY;
+        if (useAnchor) {
+            xmlpath = "anchor." + PROPERTY;
+        }
+
+        HashMap<String, String> fields = new HashMap<>();
+        try {
+            File file = new File(KitodoConfig.getKitodoConfigDirectory() + "kitodo_exportXml.xml");
+            if (file.exists() && file.canRead()) {
+                XMLConfiguration config = new XMLConfiguration(file);
+                config.setListDelimiter('&');
+                config.setReloadingStrategy(new FileChangedReloadingStrategy());
+
+                int count = config.getMaxIndex(xmlpath);
+                for (int i = 0; i <= count; i++) {
+                    String name = config.getString(xmlpath + "(" + i + ")[@name]");
+                    String value = config.getString(xmlpath + "(" + i + ")[@value]");
+                    fields.put(name, value);
+                }
+            }
+        } catch (ConfigurationException | RuntimeException e) {
+            logger.debug(e.getMessage(), e);
+            fields = new HashMap<>();
+        }
```

Comment:
Dublicated code block - see comment above.

Reply:
Thank you for noticing. The code was deduplicated now